### PR TITLE
Jupyter notebook render

### PR DESCRIPTION
**Pull request by mishaschwartz**
## Motivation and Context

 It would be nice to render jupyter notebooks... this PR does that.

## Your Changes

**Description**:

- added python dependency to render notebooks as html using nbconvert
- cache previously rendered notebooks on disk
- add helper to select range so that we can enable annotations for these files later on

**Type of change** (select all that apply):

 

- [x] New feature (non-breaking change which adds functionality)

## Testing

- added tests

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments. 
- [x] I have verified that the TravisCI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes. 
- [x] I have updated the Changelog.md file. 
- [x] I have described any required documentation changes below. 

### Required documentation changes (if applicable)

- add description of jupyter notebook conversion

### CODE REVIEW

File: app/models/submission_file.rb [hunk 1]
```diff
@@ -88,6 +90,10 @@ def is_pdf?
     File.extname(filename).casecmp('.pdf') == 0
   end
 
+  def pynb?
```
Comment (david-yz-liu) [inline app/models/submission_file.rb:93]: I'd prefer `is_pynb?` here to be consistent with the other methods.

**Review by david-yz-liu — COMMENTED** [review]
Looks great. I left a few very minor comments. I didn't review the annotation js code, as we'll enable that later.

Just a reminder that you're also adding some tests to cover some of the added lines.

File: app/models/submission_file.rb (hunk 1 above)
Comment (mishaschwartz) [reply app/models/submission_file.rb:93]: It is consistent with other methods but not with the rubocop checks. We should actually swap the rest of them over to this style or if not, then disable the cop. I'm happy with either.

Comment (david-yz-liu) [reply app/models/submission_file.rb:93]: Ah interesting, you can disable the cop then.

File: app/assets/javascripts/Components/Result/jupyter_notebook_viewer.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor() {
+    super();
+    this.state = {
+      annotations: []
+    }
+  }
+
+  getSelection = () => {
+    const iframe = document.getElementById("jupyter-notebook");
```
[the remark this answers is not in the document]
Comment (mishaschwartz) [reply app/assets/javascripts/Components/Result/jupyter_notebook_viewer.jsx:14]: sure, noted for later

**Pull request merged**